src/coreclr/vm/jithelpers.cpp

```diff
@@ -589,7 +589,12 @@ HCIMPLEND
 HCIMPL1_V(UINT64, JIT_Dbl2ULng, double val)
 {
     FCALL_CONTRACT;
+#if defined(TARGET_X86) || defined(TARGET_AMD64)
```

Comment:
Change to `if defined(TARGET_XARCH)` it covers both cases

Reply:
IIRC the VM doesn't have `TARGET_XARCH` available and we always do it explicitly instead.